internal/clock/fake_clock.go

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package clock
```

Comment:
I'd put this fake implementation in a test package just to be super clear it's a test utility

You could put it in `timetesting`